kit/tracing/tracing.go

```diff
@@ -50,15 +50,44 @@ func InjectToHTTPRequest(span opentracing.Span, req *http.Request) {
 func ExtractFromHTTPRequest(req *http.Request, handlerName string) (opentracing.Span, *http.Request) {
 	spanContext, err := opentracing.GlobalTracer().Extract(opentracing.HTTPHeaders, opentracing.HTTPHeadersCarrier(req.Header))
 	if err != nil {
-		span, ctx := opentracing.StartSpanFromContext(req.Context(), handlerName+":"+req.URL.Path)
+		span, ctx := opentracing.StartSpanFromContext(req.Context(), "request")
+		annotateSpan(span, handlerName, req.URL.Path)
+
 		LogError(span, err)
-		req = req.WithContext(ctx)
-		return span, req
+
+		return span, req.WithContext(ctx)
+	}
+
+	span := opentracing.StartSpan("request", opentracing.ChildOf(spanContext), ext.RPCServerOption(spanContext))
+	annotateSpan(span, handlerName, req.URL.Path)
+
+	return span, req.WithContext(opentracing.ContextWithSpan(req.Context(), span))
+}
+
+func annotateSpan(span opentracing.Span, handlerName, path string) {
+	span.SetTag("handler", handlerName)
+	span.SetTag("base_path", basePath(path))
+	span.LogKV("path", path)
+}
+
+// basePath derives a suitable base path
+// for example:
+// /foo/bar/1234 -> /foo
+// /api/v2/buckets/1234 -> /api/v2/buckets
+// /api/v1 -> /api
+// / -> /
+// "" -> /
+func basePath(path string) string {
```

Comment:
This doesn't seem reliable. For example:
- `/foo/bar/1234` should yield `/foo/bar`
- `/api/v2/query/suggestions` should yield `/api/v2/query/suggestions`

Reply:
Agreed I wasn't happy with what I could get out of this. Hard to say how to resolve this deterministically. Since we don't know which part of the path will be a resource type vs a resource identifier. This brings us around to your suggestion about having the router match rather than just the path. I will dig into our router a bit and see what I can find.
There are other routers that allow you that context when creating middleware.

Reply:
I don't know how the current router was chosen, but we would need to rope in more folks to switch, IMO.

> What do you think about adjusting ExtractFromHTTPRequest to take routerPath instead of handlerName?

This option would require adjusting the current pattern across all handlers, but it gives us exactly what is needed here. Ping if you would like to discuss this or other options.

Reply:
Modifying `httprouter` is great. Looks like that repo was last touched on 5-oct, so we should expect something from him.

Reply:
Yeah I would like to get us back to master ASAP.
I used the nice go mod replace so it should be a clean break switching back.